src/controllers/socialLogin.js

```diff
@@ -0,0 +1,62 @@
+import User from '../services/user.service';
+import Helper from '../helpers/helper';
+import verifyUser from '../helpers/verification-email';
+
+const login = async (req, res) => {
+  const data = req.user;
+  const firstname = data.name.givenName;
+  const lastname = data.name.middleName || data.name.familyName;
+  const email = data.emails[0].value;
+  const username = `${firstname}.${lastname}`;
+
+  // check if user is in db
+  const registeredUser = await User.findOne(email, '');
+
+  // if yes generate token
+  if (registeredUser) {
+    const payload = {
+      email,
+      role: data.role,
+      verified: data.verified
+    };
+    const token = Helper.generateToken(payload);
+
+    return res.status(200).json({
+      status: 200,
+      message: 'Logged in successfully',
+      data: {
+        firstname, lastname, username, email
+      },
+      token
+    });
+  }
+  const hasspassword = Helper.hashPassword('password');
+  const dbSchema = {
+    firstname, lastname, email, username, password: hasspassword
+  };
+  const createdUser = await User.addUser(dbSchema);
```

Comment:
@Anguandia So far you are on good progress. However, when a user is using social login and the user is not in Authors' heaven database he immediately signs up without making sure that he wants to sign up to Authors' heaven. I think it could be better if we first ask him if he is willing to proceed with sign up.

Reply:
Factored in, thanks for the observation